vault/resource_aws_secret_backend_static_role_test.go

```diff
@@ -0,0 +1,123 @@
+package vault
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/testutil"
+)
+
+func TestAccAWSSecretBackendStaticRole(t *testing.T) {
+	mount := acctest.RandomWithPrefix("tf-aws-static")
+	_, _ = testutil.GetTestAWSCreds(t)
+	resourceName := "vault_aws_secret_backend_static_role.role"
+	username := testutil.SkipTestEnvUnset(t, "AWS_STATIC_USER")[0]
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion114)
+		},
+		ProviderFactories: providerFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAWSStaticReourceConfig(mount, username),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", "test"),
```

Comment:
we can use the constants here as well
```suggestion
					resource.TestCheckResourceAttr(resourceName, consts.FieldName, "test"),
```